### DISCUSSION TITLE
Module naming conventions

### DISCUSSION THREAD

**Discussion by tiborauer** [automaticanalysis/automaticanalysis#225] [Ideas]
I guess, we should determine a naming convention for module names to avoid cases like https://github.com/rhodricusack/automaticanalysis/issues/83.

I suggest:
`aamod_'modality'_'task'[_'stream']`

**Comment by rhodricusack** [comment]
I have mixed feelings about this, because when a module has a number of
wrappers that act on different data, like
aamod_norm_write_{meanepi | cons | spmt | nodif ...}
or the many
aamod_bet_*
It seems elegant (and more memorable) to keep the connection to the
underlying module.

So my vote would be to follow Tibor's proposal, except where there are many
.xmls pointing to a single .m, where the root of the xml should be the m
filename

**Comment by tiborauer** [comment]
Now, there is also an option to change the name of the input/outputstreams on-demand, so those many .xmls differing only in data they process may not be needed any more.

**Comment by tiborauer** [comment]
As a step towards solving this issue, I am planning to rename a few modules according to the following rules:
- Modality labels and stream names should be all lowercase.
- Stream names at the end of the module name should be prefixed with an "_".
- Task/approach (e.g. ANOVA, ANTS, GIFT, fieldmap2VDM) can still be with uppercase.

These module (and other similars) will be affected:
- aamod_bet_meanEPI -> aamod_bet_meanepi
- aamod_coreg_extended_2meanepi -> aamod_coreg_extended_2_meanepi
- aamod_diffusion_coreg_extended_2DKI -> aamod_diffusion_coreg_extended_2_dki
- aamod_MTI_coreg_extended_2 -> aamod_mti_coreg_extended_2
- aamod_tSNR_EPI -> aamod_tSNR_epi

**Comment by jones-michael-s** [comment]
Hi Tibor,

Don't know if you want more to do, but here's a few other inconsistencies I've noticed:

- is "epi" plural (e.g., aamod_convert_epis.xml) or singular (e.g., aamod_bet_epi_masking.m)?
- shouldn't bet be uppercase? (e.g., aamod_BET_meanepi)
- some FSL functionality is prefaced with fsl (e.g., aamod_fsl_FAST.m) and some not (aamod_melodic.m)
- also, shouldn't "fsl" be uppercase?

Mike

**Comment by tiborauer** [comment]
To be honest, I would rather lowercase everything, however, "FAST" as "fast" would be misleading.

**Comment by tiborauer** [comment]
I am about to submit another PR refining the MEEG, updating the PPI (making it working), adding The Decoding Toolbox (great stuff!), and updating the EEGLAB interface to the new EEGLAB release. Plus some minor changes. :)

Then I jump on the module names. :)

**Comment by tiborauer** [comment]
#221
